### ISSUE TITLE
Handle pagination

### ISSUE THREAD

**Issue by mattbrailsford** [mattbrailsford/umbraco-headrest#3]
At the moment there is code to list all pages on a `/routes/` route which provides a list of all the URLs that make up the given endpoint. This currently works by going through the xml and extracting all the node ids and fetching their URLs. The problem with this is that it assumes the pages contain all their content on a single page, which if we need to use pagination, such as on a blog list, this currently wouldn't handle it.

The question then is how to handle pagination? I'm wondering if we need some kind of provider to provide all the routes for the `/routes/` which OOTB can just list all pages. Devs could then be responsible for providing the URLs in the format they need such as `/blog/2/` or something.

Just adding this as an issue for now to get feedback / suggestions from others.

**Comment by mattbrailsford** [comment]
Maybe the `RoutesListPath` option could instead become `RoutesFactory` and could be something like:

````csharp
...
RoutesFactory = new DefaultHeadRestRoutesFactory("routes")
...
````

This way the factory can say what route it uses for it's URL segment and how it builds it's routes?

**Comment by mattbrailsford** [comment]
Fixed in cbeddc73991108c30228d87a75a061a0045396d4
